bgpd/bgp_ecommunity.c

```diff
@@ -601,9 +601,12 @@ ecommunity_str2com (const char *str, int type, int keyword_included)
    ECOMMUNITY_FORMAT_ROUTE_MAP
    ECOMMUNITY_FORMAT_COMMUNITY_LIST
    ECOMMUNITY_FORMAT_DISPLAY
+
+   Filter is added to display only ECOMMUNITY_ROUTE_TARGET in some cases. 
```

Comment:
The extra parameter has been proposed to extract from a list of ecom, only the ones that are route target.
The fact is that this function is used by show running to dump the list of import-rts and export-rts.
As the list of ecom is also used to store the extended router-mac option, the show running import-rts format would bring the router-mac option. Which is not the intention.

That extra parameter acts as a filter, meaning that he assumes ecommunity_str2com is used to dump a list of router targets only. This is why the last parameter requests to extract from the list of rts only.